migrations:Update the public control container migrations

vyaghras opened the pull request:
**Description of changes:**
We need to use ReplaceStringMigration for public container. Also the old version are plain string instead of schnauzer generator string.


**Testing done:**
- Update to 1.33.0 from 1.28.0 having the old migrations as is.
```

apiclient get settings.host-containers
{
  "settings": {
    "host-containers": {
      "admin": {
        "enabled": true,
        "source": "328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-admin:v0.11.16",
        "superpowered": true,
        "user-data": "eyJzc2giOnsiYXV0aG9yaXplZC1rZXlzIjpbXX19"
      },
      "control": {
        "enabled": true,
        "source": "328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-control:v0.7.20",
        "superpowered": false
      }
    }
  }
}
bas

apiclient get settings.host-containers
{
  "settings": {
    "host-containers": {
      "admin": {
        "enabled": true,
        "source": "328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-admin:v0.11.14",
        "superpowered": true,
        "user-data": "eyJzc2giOnsiYXV0aG9yaXplZC1rZXlzIjpbXX19"
      },
      "control": {
        "enabled": true,
        "source": "328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-control:v0.7.18",
        "superpowered": false
      }
    }
  }
}
```
- Update to 1.33.0 from 1.28.0 having the old migrations updated.
```
apiclient get settings.host-containers
{
  "settings": {
    "host-containers": {
      "admin": {
        "enabled": true,
        "source": "328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-admin:v0.11.16",
        "superpowered": true,
        "user-data": "eyJzc2giOnsiYXV0aG9yaXplZC1rZXlzIjpbXX19"
      },
      "control": {
        "enabled": true,
        "source": "328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-control:v0.7.20",
        "superpowered": false
      }
    }
  }
}


apiclient get settings.host-containers
{
  "settings": {
    "host-containers": {
      "admin": {
        "enabled": true,
        "source": "328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-admin:v0.11.14",
        "superpowered": true,
        "user-data": "eyJzc2giOnsiYXV0aG9yaXplZC1rZXlzIjpbXX19"
      },
      "control": {
        "enabled": true,
        "source": "328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-control:v0.7.18",
        "superpowered": false
      }
    }
  }
}
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

vyaghras committed:
migrations: update the public control container migrations

We need to use ReplaceStringMigration for public container.
Also the old version are plain string instead of schnauzer generator
string.

cbgbt reviewed (approved):
Do we need to do some kind of catchup migration to catch customers who are stranded on `0.7.18` and `0.7.19`?

cbgbt reviewed (approved):
The code seems fine to me!

For the commit messages, make sure to add a space after `migrations:`.

vyaghras committed:
migrations: update left behind public control container

We have updated the Control host container to version 0.7.19
in Bottlerocket image version 1.30.0, but due to a miss in migration
the control container version has not been updated since
Bottlerocket version 1.29.0. This migration will update all those
containers to Control container version 0.7.19.

vyaghras committed:
migrations: update left behind public control container

We have updated the Control host container to version 0.7.20
in Bottlerocket image version 1.31.0, but due to a miss in migration
the control container version has not been updated since
Bottlerocket version 1.29.0. This migration will update all those
containers to Control container version 0.7.20.
